src/Mod/TechDraw/Gui/QGISVGTemplate.cpp

```diff
@@ -204,17 +204,30 @@ void QGISVGTemplate::updateView(bool update)
     draw();
 }
 
-void QGISVGTemplate::clearClickHandles()
+std::vector<TemplateTextField*> QGISVGTemplate::getTextFields()
 {
-    prepareGeometryChange();
-    constexpr int TemplateTextFieldType{QGraphicsItem::UserType + 160};
-    auto templateChildren = childItems();
+    constexpr int TemplateTextFieldType {QGraphicsItem::UserType + 160};
+    std::vector<TemplateTextField*> result;
+    result.reserve(childItems().size());
+
+    QList<QGraphicsItem*> templateChildren = childItems();
     for (auto& child : templateChildren) {
         if (child->type() == TemplateTextFieldType) {
-            child->hide();
-            scene()->removeItem(child);
-            delete child;
+            result.push_back(dynamic_cast<TemplateTextField*>(child));
```

Comment:
```suggestion
            result.push_back(static_cast<TemplateTextField*>(child));
```
`dynamic_cast` type checks, but we already type checked, so let's use `static_cast`

Reply:
Done :)